app/Jobs/CombineExportedPostBatchesJob.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Ushahidi\App\Jobs;
+
+use Exception;
+use RuntimeException;
+use Log;
+use Ushahidi\Core\Entity\ExportJobRepository;
+use Ushahidi\Core\Entity\ExportBatchRepository;
+use Illuminate\Http\File;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Str;
+use Illuminate\Filesystem\Filesystem;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Facades\File as LocalFilesystem;
+
+class CombineExportedPostBatchesJob extends Job
+{
+    protected $jobId;
+
+    protected $csvPrefix = 'csv/tmp';
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct($jobId)
+    {
+        $this->jobId = $jobId;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle(
+        ExportJobRepository $exportJobRepo,
+        ExportBatchRepository $exportBatchRepo
+    ) {
+        // Load job
+        $job = $exportJobRepo->get($this->jobId);
+
+        if ($job->status === 'completed') {
+            Log::debug('Job already completed', ['jobId' => $this->jobId]);
+            // All done here
+            return;
+        }
+
+        // Check if all batches exported
+        if (!$exportJobRepo->isJobFinished($this->jobId)) {
+            Log::debug('Batches not finished, requeueing', ['jobId' => $this->jobId]);
+            // Batches aren't done yet, wait 5 minutes and try again
+            dispatch(
+                (new CombineExportedPostBatchesJob($this->jobId))
+                    ->delay(Carbon::now()->addMinutes(5))
+            );
+            // All done
+            return;
+        }
+
+        // Load batches
+        $batches = $exportBatchRepo->getByJobId($this->jobId);
+        // Get just filenames
+        $fileNames = $batches
+            ->sortBy('batch_number')
+            ->sortByDesc('has_header')
+            ->pluck('filename')
+            ->all();
+
+        if (!$this->doAllFilesExist($fileNames)) {
+            Log::warning('Some files in export job do not exist', ['jobId' => $this->jobId]);
+            throw new RuntimeException('Some files in export job do not exist');
+        }
+
+        // Combine the files
+        $destinationFile = $this->combineFiles($fileNames);
+
+        // Clean up partial files
+        Storage::delete($fileNames);
+
+        // Set status = completed
+        $job->setState([
+            'url' => Storage::url($destinationFile),
```

Comment:
I'm not sure this should actually be a URL, or a filename that gets formatted to a URL in the Formatter.

Reply:
is this code meant to double for multisite and OSS? Would Storage::url be more versatile?

Reply:
OSS only right now. But it'll do both in future.
So if we store a URL then you can't move files from 1 storage driver to another, because the URL is hard coded. If we store only the path, then we can run it through Storage::url() in the formatter.

I've looked into it and we can actually make the RS driver build the Temp url for us, using a different URL for every request.